rio_tiler/io/base.py

```diff
@@ -1,18 +1,24 @@
 """rio_tiler.io.base: ABC class for rio-tiler readers."""
 
 import abc
+import re
 from dataclasses import dataclass, field
-from typing import Any, Dict, List, Optional, Sequence, Tuple
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Type, Union
 
 import numpy
 
+from ..errors import MissingAssets, TileOutsideBounds
+from ..expression import apply_expression
+from ..tasks import multi_arrays, multi_values
+from ..utils import tile_exists
+
 
 # ref: https://github.com/python/mypy/issues/5374
 @dataclass  # type: ignore
 class BaseReader(metaclass=abc.ABCMeta):
     """Rio-tiler.io BaseReader."""
 
-    bounds: Sequence[float] = field(init=False)
+    bounds: Tuple[float, float, float, float] = field(init=False)
```

Comment:
Unrelated to the PR goal.
In other methods we usually define bounds like a Tuple instead of a Sequence